### ISSUE TITLE
Bug: no-property-in-node doesn't recognize locally linked typescript-eslint types

### ISSUE THREAD

**Issue by JoshuaKGoldberg** [eslint-community/eslint-plugin-eslint-plugin#482]
Coming over from https://github.com/typescript-eslint/typescript-eslint/issues/8226 -> https://github.com/typescript-eslint/typescript-eslint/pull/9261: amusingly+ironically, the `eslint-plugin/no-property-in-node` rules doesn't detect any types in the https://github.com/typescript-eslint/typescript-eslint codebase. This is because typescript-eslint's types are defined at a path like:

```plaintext
/Users/josh/repos/typescript-eslint/packages/types/dist/generated/ast-spec.d.ts
```

...while the rule only checks `node_modules/` style paths:

https://github.com/eslint-community/eslint-plugin-eslint-plugin/blob/04e397b6d3c02084a68f5339f182fb5aecaaca5b/lib/rules/no-property-in-node.js#L3-L6

I suppose we have two possible fixes:

* Add a third path like `/packages[/\\]types[/\\]dist[/\\]generated[/\\]ast-spec\.d\.ts/,` to hardcode for typescript-eslint
* Add an option to allow user configs to specify their own paths

Proposal: can we go for the latter fix, adding a rule option? I feel awkward asking to hardcode this rule for one project. 😄

**Comment by aladdin-add** [comment]
I also agreed to adding an option.
